File: Javascript/valid-palindrome.js

function isAlphaNumeric(s){
    let reg = /^[a-zA-Z0-9]*$/;
    return s != null && reg.test(s);
}

function getCompressedString(s){
    let st = "";
    for(let i=0; i<s.length;i++){
        if(isAlphaNumeric(s[i])){
            st = st + s[i];
        }
    }
    return st.toLocaleLowerCase();
}
function reverseString(s){
    let r = "";
    for(let i=s.length-1; i>=0; i--){
        r = r + s[i];
    }
    return r;
}
function isValidPalindrome(s){
    const originalStr = getCompressedString(s);
    const reverseStr = reverseString(originalStr);
    if(originalStr === reverseStr){
        console.log("String is palindrome")
    }else{
        console.log("String is not palindrome")
    }

}

isValidPalindrome("A man, a plan, a canal: Panama")

File: ScalerTest/MathsProblems/ValidPalindrome.java
package MathsProblems;

import java.util.Locale;

public class ValidPalindrome {
    public static boolean isAlphaNumeric(String s) {
        return s != null && s.matches("^[a-zA-Z0-9]*$");
    }
    public static String getCompressedStr(String s){
        String newStr = "";
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            if(isAlphaNumeric(s.charAt(i)+"") ){
                newStr = newStr + s.charAt(i)+"";
            }
        }
        return newStr;
    }

    public static String reverseString(String s){
        String r = "";
        for(int i=s.length()-1; i>=0; i--){
            r = r + s.charAt(i);
        }
        return r;
    }
    public static boolean isPalindrome(String s){
        String originalStr = getCompressedStr(s).toLowerCase(Locale.ROOT);
        String reverseStr = reverseString(originalStr);
        if(originalStr.equals(reverseStr)){
            System.out.println("string is palindrome");
        }else{
            System.out.println("string is not palindrome");
        }
//        System.out.println(str);
        return false;
    }
    public static void main(String[] args) {
        String str = "A man, a plan, a canal: Panama";
//        String str = "race a car";
//        String str = " ";
        isPalindrome(str);
    }
}
